x-pack/plugins/observability/public/components/threshold/components/expression.tsx

```diff
@@ -54,17 +59,54 @@ export const defaultExpression = {
 } as MetricExpression;
 
 export function Expressions(props: Props) {
-  const { setRuleParams, ruleParams, errors, metadata } = props;
-  const { docLinks } = useKibana().services;
-  const { source, createDerivedIndexPattern } = useSourceContext();
+  const { setRuleParams, ruleParams, errors, metadata, onChangeMetaData } = props;
+  const { data, dataViews, dataViewEditor, docLinks } = useKibana().services;
+  const { source } = useSourceContext();
 
   const [timeSize, setTimeSize] = useState<number | undefined>(1);
   const [timeUnit, setTimeUnit] = useState<TimeUnitChar | undefined>('m');
-  const derivedIndexPattern = useMemo(
-    () => createDerivedIndexPattern(),
-    [createDerivedIndexPattern]
+  const [dataView, setDataView] = useState<DataView>();
+  const [searchSource, setSearchSource] = useState<ISearchSource>();
+  const derivedIndexPattern = useMemo<DataViewBase>(
+    () => ({
+      fields: dataView?.fields || [],
+      title: dataView?.getIndexPattern() || 'unknown-index',
+    }),
+    [dataView]
   );
 
```

Comment:
I'm still reviewing, but I wanted to mention that the preview chart is not working; it always shows "no data". I checked the other follow-up issues, but couldn't find one related to it.

I believe this should be addressed in the PR because it gives the sense that the rule is broken while listing all the fields from the DataView, and there is data.

<img width="577" alt="Screenshot 2023-06-15 at 12 51 57" src="https://github.com/elastic/kibana/assets/6838659/2d6c937e-3da0-4ca2-98fa-a0545855d612">

### Suggestion
The right solution could be done in the `source.tsx` file where the index pattern is fetched.
However, If you think this will take time, here is a workaround that makes the Preview chart works:
```suggestion
  const updatedSource = {
    ...source,
    ...{ configuration: { metricAlias: derivedIndexPattern.title } },
  };
```
Then in the `<ExpressionChart> ` use `updatedSource` instead of `source`

<img width="589" alt="Screenshot 2023-06-15 at 13 03 35" src="https://github.com/elastic/kibana/assets/6838659/23102b09-8d0a-4807-948f-9e604a0e2d91">